beacon-chain/blockchain/process_block.go

```diff
@@ -282,6 +282,14 @@ func (s *Service) onBlock(ctx context.Context, signed interfaces.ReadOnlySignedB
 	if err := s.handleEpochBoundary(ctx, postState, blockRoot[:]); err != nil {
 		return err
 	}
+	// Save state root to block root mapping
+	postStateRoot, err := postState.HashTreeRoot(ctx)
+	if err != nil {
+		return errors.Wrap(err, "could not get postStateRoot")
+	}
+	if err := s.cfg.BeaconDB.SaveBlockRoot(ctx, postStateRoot, blockRoot); err != nil {
+		return errors.Wrap(err, "could not save blockRoot")
+	}
```

Comment:
To solve the cognitive complexity lint complain, consider moving these inside `savePostStateInfo`